tokio/tests/net_panic.rs

```diff
@@ -0,0 +1,189 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+
+use std::error::Error;
+use tokio::net::{TcpListener, TcpStream};
+use tokio::runtime::{Builder, Runtime};
+
+mod support {
+    pub mod panic;
+}
+use support::panic::test_panic;
+
+#[test]
+#[cfg(not(target_os = "wasi"))]
+fn udp_socket_from_std_panic_caller() -> Result<(), Box<dyn Error>> {
```

Comment:
This needs to go on the entire file. WASM doesn't support `catch_unwind`. Sorry about that.

Reply:
Thank you, I was just digging into the Wasm32-wasi PR to work out what I was doing wrong. That saved me a bunch of time!